### PR TITLE
Basic server interoperability tests.

### PR DESCRIPTION
**Pull request by jakobr-google**
Implements the basic server interoperability tests for gRPC compliance.
Compressed tests, TLS, and authentication are not implemented yet, but
basic test cases pass against the C++ gRPC interoperability client.

Part of #10.

### CODE REVIEW

File: interop/protos/messages.proto [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+  // stream.
+  PayloadType response_type = 1;
+
+  // Configuration for each expected response message.
+  repeated ResponseParameters response_parameters = 2;
+
+  // Optional input payload sent along with the request.
+  Payload payload = 3;
+
+  // Whether server should return a given status
```
Comment (johnniwinther) [inline interop/protos/messages.proto]: `status` -> `status.`

File: interop/bin/server.dart [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  }
+}
+
+Future<Null> main(List<String> args) async {
+  final argumentParser = new ArgParser();
+  argumentParser.addOption('port', defaultsTo: '8080');
+  argumentParser.addOption('use_tls', defaultsTo: 'false');
+  final arguments = argumentParser.parse(args);
+  final port = int.parse(arguments['port']);
+  //final useTls = arguments['use_tls'] == 'true';
```
Comment (johnniwinther) [inline interop/bin/server.dart]: Remove or document why its commented out.

File: lib/src/server.dart [hunk 3]
```diff
@@ -284,13 +302,15 @@ class ServerHandler {
       final bytes = _descriptor.responseSerializer(response);
       _stream.sendData(GrpcHttpEncoder.frame(bytes));
     } catch (error) {
```
Comment (johnniwinther) [inline lib/src/server.dart]: Shouldn't we `_sinkIncoming` for this error case, like above?

File: interop/bin/server.dart (hunk 2 above)
Comment (jakobr-google) [inline interop/bin/server.dart]: Removed. I'll re-add it when I actually implements TLS handling.

File: interop/protos/messages.proto (hunk 1 above)
Comment (jakobr-google) [inline interop/protos/messages.proto]: Done.

File: lib/src/server.dart (hunk 3 above)
Comment (jakobr-google) [inline lib/src/server.dart]: Probably, yes. Done.

**Pull request merged**